### PR TITLE
Federate pinned entries

### PR DESCRIPTION
**Pull request by BentiGorlich**
This PR implements the federation of pinning and unpinning entries in magazines:

- [x] Lemmy compatibility
- [x] Mbin <-> Mbin compatibility
- [x] When a remote moderator on a local magazine pins/unpins an entry -> announce it to the subscribers
- [x] Modlog entry for when a post is pinned

### CODE REVIEW

**Comment by nobodyatroot** [comment]
testing this one next :-)

**Comment by BentiGorlich** [comment]
Same :)

**Review by nobodyatroot — DISMISSED** [review]
i can't find any issues... APPROVED!

File: src/Service/EntryManager.php [hunk 1]
```diff
@@ -267,13 +275,27 @@ public function restore(User $user, Entry $entry): void
         $this->dispatcher->dispatch(new EntryRestoredEvent($entry, $user));
     }
 
-    public function pin(Entry $entry): Entry
+    /**
+     * @param User|null $actor this should only be null if it is a system call
```
Comment (melroy89) [inline src/Service/EntryManager.php:279]: Explain in the doxygen style `@brief` section that the ping is both doing pin and unpin.

Comment (BentiGorlich) [reply src/Service/EntryManager.php:281]: done, though without the `@brief`

**Review by melroy89 — APPROVED** [review]

File: src/Service/ActivityPubManager.php [hunk 2]
```diff
@@ -512,76 +518,175 @@ public function updateMagazine(string $actorUrl): ?Magazine
...
 
-                        foreach ($moderatorsToRemove as $modToRemove) {
-                            if (null === $modToRemove) {
-                                continue;
+            $this->logger->debug('got featured items for magazine "{magName}": {json}', ['magName' => $magazine->name, 'json' => json_encode($attributedObj)]);
+
+            if (null !== $items) {
+                $pinnedToRemove = $this->entryRepository->findPinned($magazine);
+                $indexesNotToRemove = [];
+                $idsToPin = [];
```
Comment (melroy89) [inline src/Service/ActivityPubManager.php:625]: I just discovered today that `$idsToPin` is unused.

**Pull request merged**